src/compiler/checker.ts

```diff
@@ -16183,9 +16216,16 @@ namespace ts {
             return getSignatureInstantiation(signature, getInferredTypes(context), isInJavaScriptFile(contextualSignature.declaration));
         }
 
-        function inferJsxTypeArguments(signature: Signature, sourceAttributesType: Type, context: InferenceContext): Type[] {
+        function inferJsxTypeArguments(signature: Signature, node: JsxOpeningLikeElement, context: InferenceContext): Type[] {
+            // Skip context sensitive pass
+            const skipContextParamType = getTypeAtPosition(signature, 0);
+            const checkAttrTypeSkipContextSensitive = checkExpressionWithContextualType(node.attributes, skipContextParamType, identityMapper);

```

Comment:
I don’t understand this pass, or at least the name of this pass. We grab the type of the first parameter, then use it as the contextual type against node.attributes, instead of just taking the type from outside this function. But then we call getTypeAtPosition *again* — do we expect something different this time?

Reply:
Yep, we do expect something different. Namely we expect contextual parameter types to have been assigned from the first pass. We do the same thing in normal inference where we call checkExpression separately in each pass on an argument.

Reply:
Oh, right, because inferTypes is side-effecting. Or is is checkExpressionWithContextualType.

I still don't understand how this skips the context param though. Or does it skip the context[ual type] of the param?

Reply:
`checkExpressionWithContextualType` is the sideffecting one. It can cause assignments to cached parameter types when called with an inference context as the contextual mapper.